rocky/rocky/paginator.py

```diff
@@ -0,0 +1,38 @@
+from django.core.paginator import EmptyPage, Page, PageNotAnInteger, Paginator
+from django.utils.translation import gettext_lazy as _
+
+
+class RockyPaginator(Paginator):
+    def __init__(
+        self,
+        object_list,
+        per_page: int | str,
+        orphans: int = 0,
+        allow_empty_first_page: bool = True,
+    ) -> None:
+        super().__init__(object_list, per_page, orphans, allow_empty_first_page)
+
+        if orphans != 0:
+            raise ValueError("Setting orphans is not supported")
+
+    def validate_number(self, number) -> int:
+        """Validate the given 1-based page number."""
+        try:
+            if isinstance(number, float) and not number.is_integer():
+                raise ValueError
+            number = int(number)
+        except (TypeError, ValueError):
+            raise PageNotAnInteger(_("That page number is not an integer"))
+        if number < 1:
+            raise EmptyPage(_("That page number is less than 1"))
+        return number
```

Comment:
This looks almost identical to the base implementation. Why do we need to override and copy this?

Reply:
Yes, we left out the `if number > self.num_pages:
            raise EmptyPage(_("That page contains no results"))` to not do the count again, we don't need that